### PR TITLE
Refactor getting containers from docker in logs agent

### PR DESCRIPTION
**Pull request by achntrl**
### What does this PR do?

Refactor Refactor getting containers from docker in logs agent

follow up after #3367

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/logs/input/docker/client.go [hunk 1]
```diff
@@ -65,17 +66,17 @@ func computeClientAPIVersion(serverVersion string) (string, error) {
...
 // GetContainer returns the container matching id,
 // if it does not exist, returns an error.
-func GetContainer(client *client.Client, id string) (types.Container, error) {
+func GetContainer(id string) (types.ContainerJSON, error) {
 	args := filters.NewArgs()
 	args.Add("id", id)
-	containers, err := client.ContainerList(context.Background(), types.ContainerListOptions{
-		Filters: args,
-	})
+	du, err := docker.GetDockerUtil()
```
Comment (ajacquemot) [inline pkg/logs/input/docker/client.go]: nits: do we really need to allocate this all the time ?

File: pkg/logs/input/docker/client.go [hunk 2]
```diff
@@ -65,17 +66,17 @@ func computeClientAPIVersion(serverVersion string) (string, error) {
...
 	if err != nil {
-		return types.Container{}, err
+		return types.ContainerJSON{}, err
 	}
-	if len(containers) != 1 {
-		return types.Container{}, fmt.Errorf("no container matches id: %v", id)
+	container, err := du.Inspect(id, false)
```
Comment (ajacquemot) [inline pkg/logs/input/docker/client.go]: What does this false stands for ? Worth adding a comment I think

File: pkg/logs/input/docker/container.go [hunk 3]
```diff
@@ -158,10 +158,11 @@ func (c *Container) isNameMatch(nameFilter string) bool {
...
 		return false
 	}
-	for _, name := range c.container.Names {
-		if re.MatchString(name) {
-			return true
-		}
+	if len(c.container.Name) == 0 {
+		return false
+	}
+	if re.MatchString(c.container.Name) {
```
Comment (ajacquemot) [inline pkg/logs/input/docker/container.go]: nits: we could return `re.MatchString(c.container.Name)`

File: pkg/logs/input/docker/container.go [hunk 4]
```diff
@@ -175,9 +176,12 @@ func (c *Container) isLabelMatch(labelFilter string) bool {
 		parts := strings.FieldsFunc(label, func(c rune) bool {
 			return c == ':' || c == '='
 		})
+		if c.container.Config == nil {
```
Comment (ajacquemot) [inline pkg/logs/input/docker/container.go]: I feel like it's easy to forget to do this check ..

**Review by ajacquemot — APPROVED** [review]
If this is tested, let's 🚢 it, feel free to take my nits into account.

**Comment by albertvaka** [comment]
Are we getting this in for 6.12? Otherwise, please change the milestone to 6.13 🙂

**Pull request closed without merging**